spec/PeriodicBackgroundSync-index.bs

```diff
@@ -133,42 +150,54 @@ The user agent should cap the duration and frequency of these events to limit re
-    1. Set the [=periodicsync registration/time to fire=] of |registration| to |timeToFire|.
-1. Else:
-    1. Increment [=periodicsync registration/count of retries=].
-    1. If [=periodicsync registration/count of retries=] is greater than the [=maximum number of retries=] allowed, set [=periodicsync registration/count of retries=] to 0 and follow the steps to [=calculate the time to fire of the first attempt of=] the periodicSync event.
-    1. Else, Set the [=periodicsync registration/time to fire=] of |registration| to |now| + a small back-off that is exponentially proportional to [=periodicsync registration/count of retries=].
-
-## <dfn>Schedule processing</dfn> ## {#schedule-delayed-processing}
-To [=schedule processing=] of a [=periodicSync registration=] |registration|, run the following steps:
-1. Assert: |registration|'s associated [=periodicsync registration/registration state=] is [=pending=].
-1. Schedule [=fire a periodicSync event|firing a periodicSync event=] for |registration| as soon as the the device is online at or after |registration|'s [=periodicsync registration/time to fire=].
+  <div algorithm>
+    ## <dfn>Calculate a time to fire</dfn> ## {#caculate-time-to-fire}
+    This section describes how a user agent can calculate the [=periodic sync registration/time to fire=] for a [=periodic sync registration=], |registration|. [=periodic sync registration/time to fire=] value MUST be updated in response to various triggers such as registration, unregistration, change in [=online=] status, and completion of an attempt to fire a [=periodicsync event=].
+
+    Calculating the [=periodic sync registration/time to fire=] for |registration| involves running these steps:
+      1. Let |now|, a timestamp, represent the current time.
+      1. Let |origin| be the [=origin=] associated with |registration|'s [=periodic sync registration/service worker registration=].
+      1. Let |effectiveMinimumIntervalForOrigin| be the greater of [=minimum periodic sync interval for any origin=] and |registration|'s [=periodic sync registration/minimum interval=].
+      1. Increment |effectiveMinimumIntervalForOrigin| by some user agent defined amount.
+      
+         Note: User agents can use this to delay synchronization due to the user's engagement with |origin|.
+
+      1. Set [=periodic sync scheduler/effective minimum interval=] for |origin| to |effectiveMinimumIntervalForOrigin|.
+      1. Set [=periodic sync registration/time to fire=] to |now| + |effectiveMinimumIntervalForOrigin|.
+  </div>
+
+  <div algorithm>
+    ## <dfn>Process periodic sync registrations</dfn> ## {#process-periodic-sync-registrations-algorithm}
+    When the user agent starts, run the following steps [=in parallel=]:
+    1. While true:
+        1. Wait for [=minimum periodic sync interval across origins=].
+        1. Let |firedPeriodicSync| be false.
```

Comment:
This is missing the "While firedSync is false" loop from https://gist.github.com/jakearchibald/a75f658309b395983a256dbffcbc0e7b. This loop means it only  "Waits for [=minimum periodic sync interval across origins=]" if it finds a sync.

Reply:
I've added this